### PR TITLE
Improve data type coverage for Image.

### PR DESCRIPTION
**Pull request by kopylovanat**
Added type coverage for get_access() for cl_int4, cl_uint4, cl_half4
Added type coverage for Image<N, AllocatorT> for AllocatorT - not default
Added type coverage for constructors for:
 image_channel_order = (image_channel_order::a, r, rx, rg, rgx, ra, rgb, rgbx, argb, intensity, luminance, abgr)
 image_channel_type = (image_channel_type::snorm_int8, snorm_int16, unorm_short_565, unorm_short_555, unorm_int_101010)

Changed test name image_api to image_api_core to allow it to be run separately from image_api_fp16.

Remaining gap (test coverage): other accessMode for get_access(),
                               constructor image(cl_mem , const context &, event )

### CODE REVIEW

**Comment by bader** [comment]
@ProGTX, @illuhad, could you take a look, please?

**Review by bader — APPROVED** [review]

File: tests/image/image_common.h [hunk 1]
```diff
@@ -421,6 +420,383 @@ cl::sycl::vector_class<cl::sycl::byte> get_image_host(
...
+    }
+
+    // Check get_pitch()
+    if (!image_generic<dims, AllocatorT>::compare_pitch(log, img, p)) {
+      FAIL(log, "Pitches are not the same.");
+    }
+
+    // Check get_size()
+    if (img.get_size() < (numElems * elementSize)) {
+      cl::sycl::string_class message =
```
Comment (keryell) [inline tests/image/image_common.h:493]: I think you can use `std::string` for non-API related stuff.

File: tests/image/image_common.h [hunk 2]
```diff
@@ -421,6 +420,383 @@ cl::sycl::vector_class<cl::sycl::byte> get_image_host(
...
+                  img_acc.read(image_access<dims>::get_float(item), sampler);
+              (void)dataFromFloat;  // silent warning
+          };
+          cgh.parallel_for<image_kernel_read<dims>>(r, myKernel);
+          });
+
+          queue.submit([&](cl::sycl::handler &cgh) {
+          auto img_acc =
+              img2.template get_access<cl::sycl::float4,
+                                      cl::sycl::access::mode::write>(cgh);
```
Comment (keryell) [inline tests/image/image_common.h:776]: Curious indent

File: tests/image/image_common.h [hunk 3]
```diff
@@ -421,6 +420,383 @@ cl::sycl::vector_class<cl::sycl::byte> get_image_host(
...
+          auto myKernel = [expected, img_acc](cl::sycl::item<dims> item) {
+              // Write image data
+              img_acc.write(image_access<dims>::get_int(item),
+                          cl::sycl::float4(expected));
```
Comment (keryell) [inline tests/image/image_common.h:780]: Curious indent

File: tests/image/image_common.h [hunk 4]
```diff
@@ -421,6 +420,383 @@ cl::sycl::vector_class<cl::sycl::byte> get_image_host(
...
+              // Only works with a sampler
+              cl::sycl::float4 dataFromFloat =
+                  img_acc.read(image_access<dims>::get_float(item), sampler);
+              (void)dataFromFloat;  // silent warning
+          };
+          cgh.parallel_for<image_kernel_read<dims>>(r, myKernel);
+          });
+
+          queue.submit([&](cl::sycl::handler &cgh) {
+          auto img_acc =
```
Comment (keryell) [inline tests/image/image_common.h:774]: This should be more indented.

**Review by keryell — CHANGES_REQUESTED** [review]
Great improvement!

File: tests/image/image_common.h (hunk 4 above)
Comment (kopylovanat) [reply tests/image/image_common.h]: Done.

File: tests/image/image_common.h (hunk 1 above)
Comment (kopylovanat) [reply tests/image/image_common.h]: Changed to std::string.

File: tests/image/image_common.h (hunk 2 above)
Comment (kopylovanat) [reply tests/image/image_common.h]: Fixed indent.

File: tests/image/image_common.h (hunk 3 above)
Comment (kopylovanat) [reply tests/image/image_common.h]: Fixed indent.

**Review by keryell — APPROVED** [review]
Thanks!

**Comment by bader** [comment]
@ProGTX, could you take a look at this PR, please?

**Review by ProGTX — APPROVED** [review]
I'm not quite sure how `abgr` got into the SYCL 1.2.1 spec as `CL_ABGR` is an OpenCL 2.0 feature.

But the patch looks good.

**Comment by bader** [comment]
Thanks!

**Pull request merged**